heudiconv/bids.py

```diff
@@ -137,8 +137,9 @@ def tuneup_bids_json_files(json_files):
         json_basename = '_'.join(jsonfile.split('_')[:-1])
         # if we got by now all needed .json files -- we can fix them up
         # unfortunately order of "items" is not guaranteed atm
-        if len(glob(json_basename + '*.json')) == 3:
-            json_phasediffname = json_basename + '_phasediff.json'
+        json_phasediffname = json_basename + '_phasediff.json'
+        json_mag = json_basename + '_magnitude*.json'
+        if op.exists(json_phasediffname) and len(glob(json_mag)) >= 1:
```

Comment:
I wish logic was better documented, but this with `>=1` would cause saving magnitude json with a single EchoTime, whenever before it would get triggered if only both magn + phasediff was present.... is that intended?

Reply:
it has come up before
https://neurostars.org/t/fmriprep-fieldmap-phasediff-error-file-magnitude2-not-found/712

Reply:
do not really see how that relates directly, but ok ;-)